### ISSUE TITLE
Dictionary right-click drop menu error

### ISSUE THREAD

**Issue by Fiogan** [ChoicescriptIDE/main#4]
When right-clicking an underlined word to check the spelling, if one clicks elsewhere before the dictionary options load, then the right-click option doesn't work properly on misspelled words. Instead, a right-click on an underlined word just shows the usual menu instead of the dictionary options. It works properly again upon restarting the app.

**Comment by CareyJWilliams** [comment]
Confirmed, although it seems to me like it starts working again after a small period, which I am assuming is that same loading period. The longer the misspelled word is, the longer the dictionary check takes. I tried to replicate this behaviour on 'goldd' and couldn't seem to do so. I could replicate it on 'mqweqweqweqwdsdodulo' however, which takes a lot longer to return. With 'mqweqweqweqwdsdodulo' if I waited a time roughly equivalent to how long it was taking to return with (no) dictionary responses, I found right-click suggestions working again. Any chance you could try and confirm?

In short, I'm not sure that this is a behavioural bug? Perhaps more of a poor UI, the menu should be forced to remain open until it returns or the search operation should be cancelled.

**Comment by Fiogan** [comment]
Yes, confirmed; the dictionary starts working again after a short while.

Personally, I would prefer to see the search operation cancelled when the menu is closed. That way, if my computer is really lagging or if something goes wrong, I'm not stuck in an interminable search.

**Comment by CareyJWilliams** [comment]
@Fiogan another (easier?) solution here would be to have a fixed timeout. Any searches taking longer than x (5 or 10?) seconds are cancelled, regardless of whether or not they would have returned results. 

I think it's a fairly safe assumption that on most machines, results that taking that long to return are usually going to be negative anyway.

**Comment by Fiogan** [comment]
'Incomprehensibilities'—a real word, which is in the included dictionary—took 8 seconds on my faster computer.

I suppose timeout is a workable solution, since if one wanted to add a long word there's always the manual addition text box on the User Dictionary tab. Personally, though, I would much prefer to be able to click out of the search. I like the extra flexibility; I don't want to be forcibly kicked out of a legitimate search just because my computer is slow. Timeouts wouldn't be my preference, although of course they'd be manageable.

**Comment by CareyJWilliams** [comment]
That's a good point, I get similar results. The library isn't as fast as I'd like it to be... I'll see about the feasibility of cancelling a run. Thanks Fiogan!
